fv3_cap.F90

```diff
@@ -0,0 +1,1691 @@
+          if(iau_offset > 0) then
+            alarm_output_hf_ring = startTime + IAU_offsetTI
+            if( currtime > alarm_output_hf_ring ) then
+              alarm_output_hf_ring = startTime + (nhf+1_ESMF_KIND_I4)*output_interval_hf
+            endif
+          endif
+          alarm_output_hf = ESMF_AlarmCreate(clock_fv3,name='ALARM_OUTPUT_HF',  &
+                                             ringTime =alarm_output_hf_ring,    &
+                                             ringInterval =output_interval_hf,  &  !<-- Time interval between
+                                             stoptime =alarm_output_hf_stop,    &  !<-- Time interval between
+                                             ringTimeStepCount=1,               &  !<-- The Alarm rings for this many timesteps
+                                             sticky           =.false.,         &  !<-- Alarm does not ring until turned off
+                                             rc               =RC)
+          if (ESMF_LogFoundError(rcToCheck=rc, msg=ESMF_LOGERR_PASSTHRU, line=__LINE__, file=__FILE__)) return
+
+          alarm_output_ring = startTime + output_hfmax + output_interval
+        else
+          nrg = (currtime-starttime-output_hfmax)/output_interval
+          alarm_output_ring = startTime + output_hfmax + (nrg+1_ESMF_KIND_I4) * output_interval
+        endif
+      else
+        nrg = (currtime-starttime)/output_interval
+        alarm_output_ring = startTime + (nrg+1_ESMF_KIND_I4) * output_interval
+        if(iau_offset > 0) then
+          alarm_output_ring = startTime + IAU_offsetTI
+          if( currtime > alarm_output_ring ) then
+            alarm_output_ring = startTime + (nrg+1_ESMF_KIND_I4) * output_interval
+          endif
+        endif
+      endif
+
+      call ESMF_TimeIntervalSet(output_interval, h=nfhout, m=nfmout, &
+                                s=nfsout, rc=rc)
+      alarm_output = ESMF_AlarmCreate(clock_fv3, name  ='ALARM_OUTPUT',    &
+                                      ringTime         =alarm_output_ring, & !<-- Forecast/Restart start time (ESMF)
+                                      ringInterval     =output_interval,   & !<-- Time interval between
+                                      ringTimeStepCount=1,                 & !<-- The Alarm rings for this many timesteps
+                                      sticky           =.false.,           & !<-- Alarm does not ring until turned off
+                                      rc               =RC)
+      if (ESMF_LogFoundError(rcToCheck=rc, msg=ESMF_LOGERR_PASSTHRU, line=__LINE__, file=__FILE__)) return
+!
+!-----------------------------------------------------------------------
+!***  SET THE FIRST WRITE GROUP AS THE FIRST ONE TO ACT.
+!-----------------------------------------------------------------------
+!
+      n_group = 1
+!
+!end quilting
+    endif
+!
+    ! --- advertise Fields in importState and exportState -------------------
+
+    if( cpl ) then
```

Comment:
All field advertise are based on `cpl`.  In general components should always advertise what is available.  In the case of JEDI-UFS, it might be un-coupled, but fields still need to be advertised.

Reply:
I would consider JEDI-UFS is coupled (fv3 is coupled with JEDI), while standalone fv3 is not, so we need some configuration variable, if not cpl, to differentiate it.